docs/book/src/topics/crd-changes/v1alpha7-to-v1alpha8.md

```diff
@@ -0,0 +1,58 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [v1alpha7 compared to v1alpha8](#v1alpha7-compared-to-v1alpha8)
+  - [Migration](#migration)
+  - [API Changes](#api-changes)
+    - [`OpenStackCluster`](#openstackcluster)
+      - [Change to externalNetworkID](#change-to-externalnetworkid)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# v1alpha7 compared to v1alpha8
+
+> ⚠️ v1alpha8 has not been released yet.
+
+## Migration
+
+All users are encouraged to migrate their usage of the CAPO CRDs from older versions to `v1alpha8`. This includes yaml files and source code. As CAPO implements automatic conversions between the CRD versions, this migration can happen after installing the new CAPO release.
+
+## API Changes
+
+This only documents backwards incompatible changes. Fields that were added to v1alpha8 are not listed here.
+
+### `OpenStackCluster`
+
+#### Change to externalNetworkID
+
+The field `externalNetworkID` has been renamed to `externalNetwork` and isn't a string anymore but a reference to a `NetworkFilter` object.
```

Comment:
```suggestion
The field `externalNetworkID` has been renamed to `externalNetwork` and isn't a string anymore but a reference to a `NetworkFilter` object.

The field `externalNetworkID` has been renamed to `externalNetwork` and is now a `NetworkFilter` object rather than a string ID.
The `NetworkFilter` object allows selection of a network by name, by ID or by tags.
```

to mirror what we're saying in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1796